part3-mission4/src/services/article-service.js

```diff
@@ -0,0 +1,217 @@
+      pagination: { totalArticles, totalPages, currentPage: page, limit },
+    };
+  }
+
+  // 단일 게시글 조회
+  async getArticleById(articleId, userId) {
+    const include = {
+      user: { select: { username: true } },
+      comments: {
+        select: {
+          id: true,
+          content: true,
+          createdAt: true,
+          updatedAt: true,
+          user: { select: { username: true } },
+          likeCount: true,
+          ...(userId && {
+            likedBy: {
+              where: { id: userId },
+              select: { id: true, username: true },
+            },
+          }),
+        },
+      },
+      ...(userId && {
+        likedBy: {
+          where: { id: userId },
+          select: { id: true, username: true },
+        },
+      }),
+    };
+
+    const article = await prisma.article.findUnique({
+      where: { id: parseInt(articleId) },
+      include,
+    });
+
+    if (!article) {
+      throw new AppError('존재하지 않는 게시글입니다.', 404);
+    }
+
+    const articleWithLike = {
+      ...article,
+      isLiked: article.likedBy?.length > 0 || false,
+      likeCount: article.likeCount,
+      comments: article.comments.map((c) => ({
+        ...c,
+        isLiked: c.likedBy?.length > 0 || false,
+        likeCount: c.likeCount,
+      })),
+    };
+
+    return articleWithLike;
+  }
+
+  // 게시글 작성
+  async createArticle(title, content, userId) {
+    if (!userId) throw new AppError('작성자를 확인할 수 없습니다.', 400);
+
+    const newArticle = await prisma.article.create({
+      data: {
+        title,
+        content,
+        user: { connect: { id: parseInt(userId) } },
+      },
+    });
+
+    if (!newArticle) {
+      throw new AppError('게시글 등록에 실패했습니다.', 400);
+    }
+    return newArticle;
+  }
+
+  // 게시글 수정
+  async updateArticle(id, updateData, userId) {
+    const updated = await prisma.article.updateMany({
+      where: { id: parseInt(id), userId },
+      data: updateData,
+    });
+
+    if (updated.count === 0) {
+      throw new AppError('권한이 없거나 게시글이 존재하지 않습니다.', 403);
+    }
+
+    return { message: '게시글이 수정되었습니다.' };
+  }
+
+  // 게시글 삭제
+  async deleteArticle(id, userId) {
+    const deleted = await prisma.article.deleteMany({
+      where: { id: parseInt(id), userId },
+    });
+
+    if (deleted.count === 0) {
+      throw new AppError('권한이 없거나 게시글이 존재하지 않습니다.', 403);
+    }
+
+    return { message: '게시글이 삭제되었습니다.' };
+  }
+
+  // 본인이 작성한 게시글 조회
+  async getUserArticles(userId) {
+    const articles = await prisma.article.findMany({
+      where: { userId },
+      orderBy: { createdAt: 'desc' },
+      select: {
+        id: true,
+        title: true,
+        content: true,
+        tags: true,
+        images: true,
+        likeCount: true,
+      },
+    });
+    return articles;
+  }
+
+  // 좋아요한 게시글 조회
+  async getUserLikedArticles(userId) {
+    const likedArticles = await prisma.product.findMany({
+      where: { likedBy: { some: { id: parseInt(userId) } } },
+    });
```

Comment:
적절한 스키마를 통해 조회하도록 해주세요.


```suggestion
    const likedArticles = await prisma.article.findMany({
      where: { likedBy: { some: { id: parseInt(userId) } } },
    });
```